fixture-monkey-kotlin/src/test/kotlin/com/navercorp/fixturemonkey/kotlin/KotlinInnerSpecTest.kt

```diff
@@ -0,0 +1,915 @@
+                listElement(1, expected)
+            }
+            .sample()
+
+        then(actual).allMatch { it == expected }
+    }
+
+    @Property
+    fun setAfterSizeReturnsSet() {
+        val actual = SUT.giveMeBuilder(ListStringObject::class.java)
+            .setKotlinInner {
+                property("values") { size(2) }
+                property("values", ArrayList<String>())
+            }
+            .sample()
+            .values
+
+        then(actual).isEmpty()
+    }
+
+    @Property
+    fun sizeAfterSetReturnsSize() {
+        val actual = SUT.giveMeBuilder<ListStringObject>()
+            .setKotlinInner {
+                property("values", ArrayList<String>())
+                property("values") { size(2) }
+            }
+            .sample()
+            .values
+
+        then(actual).hasSize(2)
+    }
+
+    @Property
+    fun sizeAfterSetWithSeparateInnerSpecReturnsSize() {
+        val actual = SUT.giveMeBuilder<ListStringObject>()
+            .setKotlinInner {
+                property("values", ArrayList<String>())
+            }
+            .setKotlinInner {
+                property("values") { size(2) }
+            }
+            .sample()
+            .values
+
+        then(actual).hasSize(2)
+    }
+
+    @Property
+    fun setAfterSetWithSeparateInnerSpecReturnsSet() {
+        val actual = SUT.giveMeBuilder<ListStringObject>()
+            .setKotlinInner {
+                property("values") { size(2) }
+            }
+            .setKotlinInner {
+                property("values", ArrayList<String>())
+            }
+            .sample()
+            .values
+
+        then(actual).isEmpty()
+    }
+
+    @Property
+    fun innerSpecIncrementsSequence() {
+        val actual = SUT.giveMeBuilder<ListStringObject>()
+            .setKotlinInner {
+                property("values") { size(1) }
+                property("values") { size(2) }
+                property("values") { size(3) }
+            }
+            .size("values", 5)
+            .sample()
+
+        then(actual.values).hasSize(5)
+    }
+
+    @Property
+    fun setNotNull() {
+        val actual = SUT.giveMeBuilder<SimpleObject>()
+            .setKotlinInner {
+                property("str", NOT_NULL)
+            }
+            .sample()
+            .str
+
+        then(actual).isNotNull()
+    }
+
+    @Property
+    fun keysForCollection() {
+        val keyList = listOf("key1", "key2", "key3")
+
+        val actual = SUT.giveMeBuilder<MapObject>()
+            .setKotlinInner {
+                property("strMap") {
+                    keys(keyList)
+                    size(3)
+                }
+            }
+            .sample()
+            .strMap
+
+        then(actual.keys).containsAll(keyList)
+    }
+
+    @Property
+    fun valuesForCollection() {
+        val valueList = listOf("value1", "value2", "value3")
+
+        val actual = SUT.giveMeBuilder(MapObject::class.java)
```

Comment:
`SUT.giveMeBuilder<MapObect>()` might be better.

Reply:
The review comments have been incorporated in [commit](https://github.com/naver/fixture-monkey/pull/1196/commits/0e74fb083e26f2a86d75bf492839bac65472d467)